src/calculators/sg_correction.rs

```diff
@@ -1,28 +1,61 @@
-pub struct Conversions{ // TODO: replace with a global conversions, defined by a configuration initialised from main
-    temperature_units: char,
-}
-
-impl Conversions {
-    fn temperature_to_fahrenheit(&self, temp: f32) -> f32 {
-        if self.temperature_units == 'F' {
-            return temp;
-        } else if self.temperature_units == 'K' {
-            return (9.0/5.0) * temp - 459.67;
-        } else {
-            return (9.0/5.0) * temp + 32.0;
-        }
-    }
-}
-
 pub struct SgCorrection;
 
 impl SgCorrection {
     pub fn correct_sg(&self, sg: f32, calibration_temperature: f32, measured_temperature: f32) -> f32 {
-        let conversions = Conversions{temperature_units: 'C'}; // TODO: replace with global conversions
-        let ctf = conversions.temperature_to_fahrenheit(calibration_temperature);
-        let mtf = conversions.temperature_to_fahrenheit(measured_temperature);
+        let conversions = Temperature::Celsius; // TODO: replace with global conversions
+        let ctf = conversions.to_farenheight(calibration_temperature);
+        let mtf = conversions.to_farenheight(measured_temperature);
 
         return sg * (  (1.00130346 - 0.000134722124 * mtf + 0.00000204052596 * (mtf * mtf) - 0.00000000232820948 * (mtf * mtf * mtf))
                      / (1.00130346 - 0.000134722124 * ctf + 0.00000204052596 * (ctf * ctf) - 0.00000000232820948 * (ctf * ctf * ctf)));
     }
 }
+
+pub enum Temperature {
+    Celsius,
+    Farenheight,
+    Kelvin,
+}
+
+pub struct TempParseError;
+
+impl std::convert::TryFrom<char> for Temperature {
```

Comment:
Actually I did this yesterday using the measurements dependency. Benefits of using is that there is also conversions for volumes, weights, etc. (See #25 )

Reply:
OK, I'll go through and make it compatiable with the measurements dependency

Reply:
This implementation looks better than the initial one in #25 though. Can you modify the now merged units conversion structs after the rebase so they use similar pattern?

Reply:
The #25 uses units from the measurements dependency. Only thing done in the conversions.rs is to try and figure out the unit from string. I think you should remove these and use the measurements::Temperature struct instead

Reply:
So should I just remove what I did in sg_correction.rs, and instead just use the measurements::Temperature struct?

Reply:
That's one way to do it, also if you feel like changing `conversions.rs` to use `TryFrom` instead, while still using measurements::Temperature please go ahead (can be another PR though)

Reply:
Ok, I'll change it to be `TryFrom`, then it'll be ready to merge?

Reply:
sounds like a plan

Reply:
I couldn't see an ergonomic way to change it to `TryFrom`, and have instead have decided to make all the conversion methods return a `Result<Self, ParseFloatError>`, so error handling can be called where the function is called, rather than an immediate `.unwrap()`, upon the parsing failing